src/v/raft/consensus.cc

```diff
@@ -1383,10 +1381,32 @@ ss::future<> consensus::do_start() {
           "Configuration manager started: {}",
           _configuration_manager);
 
-        co_await _offset_translator.start(
-          offset_translator::must_reset{initial_state});
-
-        co_await _snapshot_lock.with([this] { return hydrate_snapshot(); });
+        std::optional<storage::truncate_prefix_config> start_truncate_cfg;
+        auto metadata = co_await read_snapshot_metadata();
+        if (metadata.has_value()) {
+            load_from_metadata(metadata.value());
+            co_await _configuration_manager.add(
+              _last_snapshot_index, std::move(metadata->latest_configuration));
+            _probe->configuration_update();
+
+            auto delta = delta_for_truncation(metadata.value());
+            if (delta.has_value()) {
+                start_truncate_cfg = storage::truncate_prefix_config(
+                  model::next_offset(_last_snapshot_index),
+                  _scheduling.default_iopc,
+                  delta);
+
+                _flushed_offset = std::max(
+                  _last_snapshot_index, _flushed_offset);
+                co_await _configuration_manager.prefix_truncate(
+                  _last_snapshot_index);
+            }
+            _snapshot_size = co_await _snapshot_mgr.get_snapshot_size();
+        }
+        // It's expected that the log has already been initialized and
+        // truncated with metadata from the snapshot. We may still need to
```

Comment:
I'm confused by this comment. Where has the log "been already truncated with metadata from the snapshot"? We are doing it here for the first time AFAICS...

Reply:
Ah oops, this is an artifact of an older implementation that truncated earlier. Removed